src/libaktualizr-posix/ipuptanesecondary.h

```diff
@@ -28,35 +26,39 @@ class IpUptaneSecondary : public SecondaryInterface {
   void init(std::shared_ptr<SecondaryProvider> secondary_provider_in) override {
     secondary_provider_ = std::move(secondary_provider_in);
   }
-  bool putMetadata(const Target& target) override;
+  data::InstallationResult putMetadata(const Target& target) override;
   int32_t getRootVersion(bool /* director */) const override { return 0; }
-  bool putRoot(const std::string& /* root */, bool /* director */) override { return true; }
+  // TODO(OTA-4552): Support multiple Root rotations.
+  data::InstallationResult putRoot(const std::string& /* root */, bool /* director */) override {
+    return data::InstallationResult(data::ResultCode::Numeric::kOk, "");
+  }
   Manifest getManifest() const override;
   bool ping() const override;
-  data::ResultCode::Numeric sendFirmware(const Uptane::Target& target) override;
-  data::ResultCode::Numeric install(const Uptane::Target& target) override;
+  data::InstallationResult sendFirmware(const Uptane::Target& target) override;
+  data::InstallationResult install(const Uptane::Target& target) override;
 
  private:
   const std::pair<std::string, uint16_t>& getAddr() const { return addr_; }
-  bool putMetadata_v1(const Uptane::MetaBundle& meta_bundle);
-  bool putMetadata_v2(const Uptane::MetaBundle& meta_bundle);
-  data::ResultCode::Numeric sendFirmware_v1(const Uptane::Target& target);
-  data::ResultCode::Numeric sendFirmware_v2(const Uptane::Target& target);
-  data::ResultCode::Numeric install_v1(const Uptane::Target& target);
-  data::ResultCode::Numeric install_v2(const Uptane::Target& target);
+  void getSecondaryVersion() const;
```

Comment:
What is the benefit of declaring this `const` and `protocol_version` `mutable`?

Reply:
Yeah, it's annoying, I know. The problem is that `getManifest` is const, and that call this. The options are either this or changing `getManifest` in all Secondary implementations to non-const. What's your preference?

Reply:
Ah, I see. I think it is fine to leave it like this.